flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -398,6 +398,34 @@ public void alterDatabase(
         }
     }
 
+    @Override
+    public void renameDatabase(
+            String databaseName, String newDatabaseName, boolean ignoreIfNotExists)
+            throws DatabaseNotExistException, DatabaseAlreadyExistException, CatalogException {
+        checkArgument(
+                !isNullOrWhitespaceOnly(databaseName), "databaseName cannot be null or empty");
+        checkArgument(
+                !isNullOrWhitespaceOnly(newDatabaseName),
+                "newDatabaseName cannot be null or empty");
+
+        try {
+            if (databaseExists(databaseName)) {
+                if (databaseExists(newDatabaseName)) {
+                    throw new DatabaseAlreadyExistException(getName(), newDatabaseName);
+                } else {
+                    Database database = getHiveDatabase(databaseName);
+                    database.setName(newDatabaseName);
+                    client.alterDatabase(databaseName, database);
```

Comment:
Hey @JingsongLi , thanks for assigning this ticket to me!
I implemented `renameDatabase` function in `HiveCatalog`, however, [one of the tests I added](https://github.com/apache/flink/pull/17831/files#diff-3085e843995d9bd0b96921113b3eb99625d8fe4f187b3c2639cb7c2845ad373bR274) is failing, seems `client.alterDatabase(databaseName, database)` is not working as expected. I'm not sure if there's anything else I could do to update the database name, would you mind taking a look to see if I made any mistakes in this implementation?
Thanks for your help!

Reply:
Hi @shenzhu , according to source code of version [hive-2.1.1](https://github.com/apache/hive/blob/rel/release-2.1.1/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L766), this method `alterDatabase` supports only the parameters of the database or the owner can be changed.
Maybe you can consiter to drop the database firstly and then create a new one.

Reply:
Hey @YesOrNo828 , thanks for your suggestion!
However, in the current design seems we [can not drop database that's not empty](https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java#L435). Sorry I'm not very familiar with hive, do you think it's valid to have the 1-1 mapping between [InvalidOperationException](https://hive.apache.org/javadocs/r2.1.1/api/org/apache/hadoop/hive/metastore/api/InvalidOperationException.html) and [DatabaseNotEmptyException](https://nightlies.apache.org/flink/flink-docs-release-1.10/api/java/org/apache/flink/table/catalog/exceptions/DatabaseNotEmptyException.html)?

Reply:
@shenzhu  you can set param `cascade = true`, when drop a not empty database by this [method](https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java#L426).

Reply:
Hey @YesOrNo828 , thanks for your suggestion! Sorry I didn't realized that before, I just updated this PR.